compiler/optimizer/OMRValuePropagation.cpp

```diff
@@ -7355,6 +7358,33 @@ void OMR::ValuePropagation::doDelayedTransformations()
          }
       _needMultiLeafArrayCopy.deleteAll();
       }
+
+   if (TR::Compiler->om.areFlattenableValueTypesEnabled())
+      {
+      ListIterator<TR_NeedRuntimeCheckNullRestrictedArrayCopy> tt(&_needRuntimeCheckNullRestrictedArrayCopy);
+      TR_NeedRuntimeCheckNullRestrictedArrayCopy *nullRestrictedArrayCopyTree;
+
+      for (nullRestrictedArrayCopyTree = tt.getFirst();
+           nullRestrictedArrayCopyTree; nullRestrictedArrayCopyTree = tt.getNext())
+         {
+         if (trace())
+            {
+            TR::CFG *cfg = comp()->getFlowGraph();
+            comp()->dumpMethodTrees("Trees before transformNullRestrictedArrayCopy");
+            comp()->getDebug()->print(comp()->getOutFile(), cfg);
+            }
+
+         transformNullRestrictedArrayCopy(nullRestrictedArrayCopyTree);
+
+         if (trace())
+            {
+            TR::CFG *cfg = comp()->getFlowGraph();
+            comp()->dumpMethodTrees("Trees after transformNullRestrictedArrayCopy");
+            comp()->getDebug()->print(comp()->getOutFile(), cfg);
+            }
```

Comment:
I noticed after making this comment that other transformations that are related to `arraycopy` also dump trees after every transformation.  I wonder whether it would be appropriate to introduce a way of requesting a more detailed dump for Value Propagation, and placing all of these individual calls to `dumpMethodTrees` under that option.  Obviously that would be a bigger change that would be for discussion, and shouldn't be done under this pull request.

Reply:
> it would be sufficient to dump the trees just before the first arraycopy transformation and after the last

Because there are other sub arraycopy transformations happen before `transformNullRestrictedArrayCopy`, the trees have changed quite a bit compared to the initial trees before the first arraycopy transformation. Dumping before and after trees here is useful to understand what `transformNullRestrictedArrayCopy` does here.

Reply:
https://github.com/eclipse-openj9/openj9/issues/17957 is created